src/editors/containers/VideoUploadEditor/hooks.test.js

```diff
@@ -0,0 +1,68 @@
+import hooks from './hooks';
+import * as requests from '../../data/redux/thunkActions/requests';
+
+jest.mock('../../data/redux/thunkActions/requests');
+
+describe('uploadVideo', () => {
+  const dispatch = jest.fn();
+  const supportedFiles = [
+    new File(['content'], 'file1.mp4', { type: 'video/mp4' }),
+    new File(['content'], 'file2.mov', { type: 'video/quicktime' }),
+  ];
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should dispatch uploadVideo action with correct data and onSuccess callback', () => {
+    const data = {
+      files: [
+        { file_name: 'file1.mp4', content_type: 'video/mp4' },
+        { file_name: 'file2.mov', content_type: 'video/quicktime' },
+      ],
+    };
+    const onSuccess = jest.fn();
+    hooks.uploadVideo({ dispatch, supportedFiles });
+    expect(dispatch).toHaveBeenCalledWith(requests.uploadVideo({ data, onSuccess }));
```

Comment:
I have doubts this works as intended.

Reply:
@pkulkark As far as I know, `requests.uploadVideo({ data, onSuccess })` is going to return `undefined`, since it's mocked, so it doesn't really test anything. I noticed that this pattern is used all over this code base, and imho it's a bad pattern and should be avoided.

Reply:
@Cup0fCoffee Wouldn't the test fail in that case? I wanted to mock the studio API call. I admin my js testing skills is not very good. How would you suggest this be changed?

Reply:
@pkulkark You can test it by replacing the `requests.uploadVideo({ data, onSuccess })` with `undefined` and running the tests - they will pass, because the mock doesn't have an implementation.